### PR TITLE
Zetia/add component jobs workflow

### PR DESCRIPTION
**Pull request by zetiaatgithub**
# PR into Azure/azureml-examples

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

**Review by Zhong-J — APPROVED** [review]

File: .github/kubernetes-compute/convert.py [hunk 1]
```diff
@@ -21,30 +23,50 @@ def convert(input_file, compute_target, instance_type, common_runtime, output_fi
...
-                for step in data.get("jobs", {}):
-                    data["jobs"][step]["compute"] = "azureml:%s" % compute_target
-
         # set instance type
         if not is_pipeline_job and instance_type:
             resources = data.get("resources", {})
             resources["instance_type"] = instance_type
             data["resources"] = resources
-
+
```
Comment (snowei) [inline .github/kubernetes-compute/convert.py:31]: spaces.

File: .github/workflows/kubernetes-compute-cpu-resources-cleanup.yml [hunk 2]
```diff
@@ -58,36 +53,27 @@ jobs:
 
     # cleanup resources
     - name: delete_endpoints
-      if: ${{ always() }}
+      if: ${{ always() && github.event.inputs.DELETE_ENDPOINTS == 'true' }}
```
Comment (snowei) [inline .github/workflows/kubernetes-compute-cpu-resources-cleanup.yml:56]: do we still need always?

File: .github/kubernetes-compute/convert.py (hunk 1 above)
Comment (zetiaatgithub) [reply .github/kubernetes-compute/convert.py]: fixed by running black

File: .github/workflows/kubernetes-compute-cpu-resources-setup.yml [hunk 3]
```diff
@@ -97,41 +93,21 @@ jobs:
...
-      timeout-minutes: 300
-
-    - name: remove_lock_file
-      run: |
-        bash .github/kubernetes-compute/tool.sh remove_lock_file
-      timeout-minutes: 30
-
     - name: file_icm
-      if: ${{ always() }}
+      if: ${{ failure() && github.event_name != 'pull_request' }}
```
Comment (snowei) [inline .github/workflows/kubernetes-compute-cpu-resources-setup.yml:97]: also for manual trigger?

File: .github/workflows/kubernetes-compute-cpu-resources-cleanup.yml (hunk 2 above)
Comment (zetiaatgithub) [reply .github/workflows/kubernetes-compute-cpu-resources-cleanup.yml:56]: yes. always() means ignoring the status of previous steps. and I also set separate github.event.inputs.DELETE_ENDPOINTS flag for each step to manually control which step to run.

**Review by snowei — COMMENTED** [review]
please check if condition.

File: .github/workflows/kubernetes-compute-cpu-resources-setup.yml (hunk 3 above)
Comment (zetiaatgithub) [reply .github/workflows/kubernetes-compute-cpu-resources-setup.yml:97]: yes.

**Review by snowei — APPROVED** [review]

**Pull request merged**